internal/worldstate/leveldb/commit_and_query.go

```diff
@@ -163,6 +165,84 @@ func (l *LevelDB) GetIndexDefinition(dbName string) ([]byte, *types.Metadata, er
 	return l.Get(worldstate.DatabasesDBName, dbName)
 }
 
+// Iterator iterates over a range of keys
+type Iterator struct {
+	iterator.Iterator
+}
```

Comment:
w.r.t. to a prior comment: we could add a field `marshalErr error` and override Error() to return all the errors the same way, I think. I mean return and error either from the `marshalErr` field or from `iterator.Iterator.Error()`?

```
type Iterator struct {
	iterator.Iterator
	marshalErr error
}

func (i *Iterator) Error() error {
	iErr := i.Iterator.Error()
	if iErr != nil {
		return iErr
	}
	return i.marshalErr
}
```

Reply:
Iterator error is associated with the `Next()`, `Prev()`, etc... only. From my Fabric experience, iterator error would occur only when `Next()` or other iterator method is called on a iterator that holds a closed database.

Reply:
also when we call any seek method after closing the iterator.

Reply:
to avoid this confusion, let me move the `*ValueAndMetadata` to the `types` package and reuse the leveldb iterator itself.